x/pricefeed/keeper/keeper.go

```diff
@@ -227,16 +207,27 @@ func (k Keeper) GetCurrentPrices(ctx sdk.Context) types.CurrentPrices {
 }
 
 // GetRawPrices fetches the set of all prices posted by oracles for an asset
-func (k Keeper) GetRawPrices(ctx sdk.Context, marketID string) (types.PostedPrices, error) {
-	store := ctx.KVStore(k.key)
-	bz := store.Get(types.RawPriceKey(marketID))
-	if bz == nil {
-		return types.PostedPrices{}, nil
-	}
-	var prices types.PostedPrices
-	err := k.cdc.UnmarshalBinaryBare(bz, &prices)
-	if err != nil {
-		return types.PostedPrices{}, err
+func (k Keeper) GetRawPrices(ctx sdk.Context, marketId string) []types.PostedPrice {
+	var pps []types.PostedPrice
+	k.IterateRawPrices(ctx, func(pp types.PostedPrice) (stop bool) {
+		if pp.MarketId == marketId {
```

Comment:
Store iterators iterate through all keys in the store with a certain prefix. Since you have the raw price key as `RawPriceFeedPrefix<market id><oracle>` you could create an iterator for `RawPriceFeedPrefix<market id>` which would return the prices just for one market. It avoids unmarshalling and checking market ID on all prices.

Swap does this in `IterateDepositorSharesByOwner`.

Be careful with variable length keys though. Iterating over "btcb:usd" would also iterate over the prices under "btcb:usd:30" which is a different market. This can be fixed by converting market ids to a fixed length of bytes in the key.

I'll leave as a nit (your discretion on implementing or not) as I don't think the change gives us anything. Any performance increase is likely tiny given how few markets we have.